mailbox/backup/src/main/java/org/apache/james/mailbox/backup/Zipper.java

```diff
@@ -115,19 +121,27 @@ private void storeInArchive(MailboxAnnotation annotation, String directory, ZipA
         archiveOutputStream.closeArchiveEntry();
     }
 
-    private void storeInArchive(MailboxMessage message, ZipArchiveOutputStream archiveOutputStream) throws IOException {
+    private void storeInArchive(MessageResult message, ZipArchiveOutputStream archiveOutputStream) throws IOException {
         String entryId = message.getMessageId().serialize();
         ZipArchiveEntry archiveEntry = (ZipArchiveEntry) archiveOutputStream.createArchiveEntry(new File(entryId), entryId);
 
-        archiveEntry.addExtraField(new SizeExtraField(message.getFullContentOctets()));
+        archiveEntry.addExtraField(new SizeExtraField(message.getSize()));
         archiveEntry.addExtraField(new UidExtraField(message.getUid().asLong()));
         archiveEntry.addExtraField(new MessageIdExtraField(message.getMessageId().serialize()));
         archiveEntry.addExtraField(new MailboxIdExtraField(message.getMailboxId().serialize()));
         archiveEntry.addExtraField(new InternalDateExtraField(message.getInternalDate()));
-        archiveEntry.addExtraField(new FlagsExtraField(message.createFlags()));
+        archiveEntry.addExtraField(new FlagsExtraField(message.getFlags()));
 
         archiveOutputStream.putArchiveEntry(archiveEntry);
-        IOUtils.copy(message.getFullContent(), archiveOutputStream);
+        try {
+            Content  content = message.getFullContent();
+            try (InputStream stream = content.getInputStream()) {
+                IOUtils.copy(stream, archiveOutputStream);
+            }
+        } catch (MailboxException e) {
+           LOGGER.error("Error while storing message in archive", e);
```

Comment:
an error during the stream copy means it's probably partially written. How will it be handled on restore?

Reply:
I have not yet handle this case during restore, but the message will be ignored. And for the moment an error will be logged